src/api/user/RoleChannel.js

```diff
@@ -0,0 +1,52 @@
+import { BROADCAST_CHANNEL_NAME } from './constants';
+
+class RoleChannel {
+    constructor(openmct, channelName = BROADCAST_CHANNEL_NAME) {
+        this.openmct = openmct;
+        this.channelName = channelName;
+        this.roleChannel = undefined;
```

Comment:
We should make these [private fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), e.g.: `this.#roleChannel`. I don't think we need to set it to `undefined` though do we?